pages/recipe/[id]/index.js

```diff
@@ -0,0 +1,162 @@
+import IconButton from "@/components/Styled/IconButton";
+import Image from "next/image";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+import styled from "styled-components";
+import Link from "next/link";
+import { useState } from "react";
+
+export default function DetailPage() {
+  const [content, setContent] = useState("instructions");
+  const router = useRouter();
+  const { id } = router.query;
+
+  const { data: recipe, isLoading, error } = useSWR(`/api/recipes/${id}`);
+
+  if (error) {
+    return <h1>error</h1>;
+  }
+  if (isLoading) {
+    return <h1>loading recipe...</h1>;
+  }
+  if (!recipe) {
+    return <h1>loading recipe...</h1>;
+  }
```

Comment:
this would save you 3 lines of code:

```suggestion
  if (isLoading || !recipe) {
    return <h1>loading recipe...</h1>;
  }
```